tasks/main.yml

```diff
@@ -2,16 +2,58 @@
   timezone:
     name: "{{ koha_server_timezone }}"
 
-- name: Add Koha package repository signing key
+- name: Check to have {{ koha_domain }} domain in /etc/hosts
+  become: yes
+  lineinfile:
+    path: /etc/hosts
+    line: "127.0.1.1 {{ koha_domain }}"
+    insertafter: '127\.0\.0\.1 localhost'
+    state: present
+
+
+
+- name: Add Koha package community repository signing key
   apt_key:
     keyserver: "{{ koha_keyserver }}"
     id: "{{ koha_signing_key_id }}"
+  when: koha_signing_key_id is defined
+
+- name: Add Koha package community repository signing key by content
+  apt_key:
+    data: "{{ koha_signing_key_content }}"
+  when: koha_signing_key_content is defined
 
-- name: Add Koha package repository
+- name: Add Koha package community repository
   apt_repository:
     repo: "{{ koha_package_repository }}"
     state: present
-    filename: koha-packages
+    filename: koha-community
+
+- name: Add Koha package our repository signing key
+  apt_key:
+    keyserver: "{{ koha_extra_keyserver }}"
+    id: "{{ koha_extra_signing_key_id }}"
+  when: koha_extra_signing_key_id is defined and koha_extra_keyserver is defined
+
+- name: Add Koha package our repository signing key by content
+  apt_key:
+    data: "{{ koha_extra_signing_key_content }}"
+  when: koha_extra_signing_key_content is defined
+
+- name: Add Koha package our repository
+  apt_repository:
+    repo: "{{ koha_extra_package_repository }}"
```

Comment:
extra -> secondary. Or if you prefer maybe all instances of secondary I mentioned above should also be extra? Either word is fine in my opinion, but just lets keep it consistent.